helm/cluster-vsphere/templates/kubeadmcontrolplane.yaml

```diff
@@ -1,105 +1,186 @@
+      patches:
+        directory: "/tmp/kubeadm/patches"
+      nodeRegistration:
+        criSocket: /run/containerd/containerd.sock
+        kubeletExtraArgs:
+          {{- include "kubeletExtraArgs" . | nindent 10}}
+    joinConfiguration:
+      patches:
+        directory: "/tmp/kubeadm/patches"
+      nodeRegistration:
+        criSocket: /run/containerd/containerd.sock
+        kubeletExtraArgs:
+          {{- include "kubeletExtraArgs" . | nindent 10}}
+    files:
+      {{- if $.Values.proxy.enabled }}
+      {{- include "containerdProxyConfig" . | nindent 6}}
+      {{- end }}
+      {{- include "kubeProxyFiles" . | nindent 6 }}
+      {{- range $kubeadmPatch, $_ :=  .Files.Glob  "files/etc/patches/**" }}
+      - path: {{ (printf "/tmp/kubeadm/patches/%s" (base $kubeadmPatch)) }}
+        content: |-
+          {{- $.Files.Get $kubeadmPatch | nindent 10 }}
+      {{- end }}
+      {{- if $.Values.cluster.enableEncryptionProvider }}
+      - path: /etc/kubernetes/encryption/config.yaml
+        permissions: "0600"
+        contentFrom:
+          secret:
+            name: {{ include "resource.default.name" $ }}-encryption-provider-config
+            key: encryption
+      {{- end }}
+      - content: |
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            creationTimestamp: null
+            name: kube-vip
+            namespace: kube-system
+          spec:
+            containers:
+            - args:
+              - start
+              env:
+              - name: vip_arp
+                value: "true"
+              - name: vip_leaderelection
+                value: "true"
+              - name: vip_address
+                value: {{ .Values.network.controlPlaneEndpoint.host }}
+              - name: vip_interface
+                value: eth0
+              - name: vip_leaseduration
+                value: "15"
+              - name: vip_renewdeadline
+                value: "10"
+              - name: vip_retryperiod
+                value: "2"
+              image: ghcr.io/kube-vip/kube-vip:v0.3.5
+              imagePullPolicy: IfNotPresent
+              name: kube-vip
+              resources: {}
+              securityContext:
+                capabilities:
+                  add:
+                  - NET_ADMIN
+                  - SYS_TIME
+              volumeMounts:
+              - mountPath: /etc/kubernetes/admin.conf
+                name: kubeconfig
+            hostNetwork: true
+            volumes:
+            - hostPath:
+                path: /etc/kubernetes/admin.conf
+                type: FileOrCreate
+              name: kubeconfig
+          status: {}
+        owner: root:root
+        path: /etc/kubernetes/manifests/kube-vip.yaml
+    preKubeadmCommands:
+      # - bash /tmp/kubeadm/patches/kube-apiserver-patch.sh {{ .Values.controlPlane.resourceRatio }}
+      # - bash /run/kubeadm/gs-kube-proxy-patch.sh
+      - hostname  '{{ "{{" }} ds.meta_data.hostname {{ "}}" }}'
```

Comment:
Can you move this logic to a separate helper function and use it in both kubeadmconfig and kubeadmcontrolplane?

Reply:
done in https://github.com/giantswarm/cluster-vsphere/pull/23/commits/4dc89741dda5ef3ccedee527736715a9176e7ced